tools/p4mlir-export-bmv2/p4mlir-export-bmv2.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <cstdlib>
+#include <iostream>
+#include "options.h"
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+#include "mlir/IR/OperationSupport.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Parser/Parser.h"
+#include "p4mlir/Dialect/P4HIR/P4HIR_Dialect.h"
+#pragma GCC diagnostic pop
+
+#include "json_emitter.h"
+
+
+int main(int argc, char *const argv[]) {
+
+    P4::AutoCompileContext autoP4MLIRExportContext(new P4::MLIR::ExportContext);
```

Comment:
This should be entirely MLIR / LLVM-based tool, therefore no P4C code should be imported / used.

Reply:
Thanks, let me change it.

Reply:
Overall, try not to blindly copy & paste pieces from different places.

Reply:
This is the draft version, which will later be cleaned out.

Reply:
> Overall, try not to blindly copy & paste pieces from different places.

Oh, this code is useful; without it, the test will not pass.

Reply:
Well, maybe you can do one step further? And try to figure out why "the test will not pass" and fix the code afterwards? Provided that your input is pure MLIR.

Reply:
Okay, got you.

Reply:
>This should be entirely MLIR / LLVM-based tool, therefore no P4C code should be imported / used.

Are you sure? Don't we still need the parser/frontend for the P4 BMv2 program?

Reply:
Why? The input should be MLIR (in P4HIR dialect) as produced by `p4mlir-translate`. So, no frontend / parser here, only "dialect conversion".

Reply:
Ah, missed the context in the original issue (https://github.com/p4lang/p4mlir-incubator/issues/151#issuecomment-2768081285). Makes sense then.

Reply:
@Joejiong Anything under the `P4::` namespace comes from P4C. If your input is a `.mlir` program, you won't need that. I'd suggest that you take a look at [`circt-translate`](https://github.com/llvm/circt/blob/main/tools/circt-translate/circt-translate.cpp) and follow the `--export-systemc` option to learn how to build the MLIR binary we need here.